Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -72,47 +69,51 @@ export function createProtocolFilter(): Middleware {
             }
         },
         didChange: async (textDocumentChangeEvent, sendMessage) => {
+            await clients.ActiveClient.awaitUntilLanguageClientReady();
             const me: Client = clients.getClientFor(textDocumentChangeEvent.document.uri);
-            me.onDidChangeTextDocument(textDocumentChangeEvent);
-            sendMessage(textDocumentChangeEvent);
+            if (me.TrackedDocuments.has(textDocumentChangeEvent.document)) {
```

Comment:
Shouldn't every file be tracked already?  It seems dangerous to ever skip processing a didChange. Should this be an assert instead?

Reply:
We may defer 'didOpen'.  We may not consider the file opened until a `onDidChangeVisibleTextEditors` fires (or, apparently, when we receive a document symbol request for it).  Likely due to asyncs, we may receive a didChange before we finish processing a delayed didOpen.  That's perfectly fine, as we deliver the current state of the file to the native side when we do, and track changes from that point on.  This check avoids sending unnecessary didChange messages to the native side before we've gotten around to considering the file open.  Previously, these would not get discarded until a similar check occurred on the native side, after receiving them.  There is a comment to this effect in the native code, in `textDocument_didChange`, where it exits early when `get_document` does not find the document.

Reply:
That said, I removed it, just in case...